modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java

```diff
@@ -242,7 +244,17 @@
         filters.put("dictionary_decompounder", requiresAnalysisSettings(DictionaryCompoundWordTokenFilterFactory::new));
         filters.put("dutch_stem", DutchStemTokenFilterFactory::new);
         filters.put("edge_ngram", EdgeNGramTokenFilterFactory::new);
-        filters.put("edgeNGram", EdgeNGramTokenFilterFactory::new);
+        filters.put("edgeNGram", (IndexSettings indexSettings, Environment environment, String name, Settings settings) -> {
+            if (indexSettings.getIndexVersionCreated().onOrAfter(org.elasticsearch.Version.V_7_6_0)) {
```

Comment:
Unfortunately we can only error for these cases starting with 7.6 now since there might already be existing indices <7.6 that we don't want to break on minor version upgrade. Those should at least be logging a deprecation now.

Reply:
Do you think we should wait until 8 because this is technically breaking? I mean, using them relies on a bug, but still.

Reply:
We already removed the PreConfiguredTokenFilter registration for "nGram" and "edgeNGram" in 8 (using those should have thrown errors since 7.0), but I see we still register them in `getTokenFilters()`. Still we clearly [documented the deprecation and removal](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#deprecated-ngram-edgengram-token-filter-cannot-be-used) and I think we don't mention those variants in the docs for a long time now. I think disallowing new index creation using those in 7.6 should be okay, wdyt?
I think for 8.0 this means we cannot remove the registration for these names just yet and need to forward-port this error/deprecation handling there as well unfortunately.

Reply:
I just don't want someone who does a minor upgrade to have their daily indices not get created any more.

Reply:
Sound like we should start logging the deprecation warning with 7.6 and throw an error on 8.0 for this then. That would only apply for this type of custom filter use left out by the original deprecation back in https://github.com/elastic/elasticsearch/pull/30209.
I will update the PR and see if I can change the target branch so we add this on master and then backport to 7.x